### PR TITLE
Issue 280

### LINKED ISSUE #280: [Feature]: add imagePullSecret to k8sgpt crd spec
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

No

### Problem Description

from this [pr](https://github.com/k8sgpt-ai/k8sgpt-operator/pull/272) users can use other repositories (e.g. their own image) but imagePullSecrets does not added, so that user cannot use private repository.

### Solution Description

add imagePullSecret name. and if it is given, add it k8sgpt serviceaccount `imagePullSecret`.

### Benefits

users can use private repository

### Potential Drawbacks

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by JuHyung-Son**
Closes #280  

## 📑 Description

add imagePullSecrets to k8sgpt spec.
if given, imagePullSecrets will be added to `k8sgpt` serviceaccounts

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

i couldn't test it on my local environment. please test it works well. i will ask how to set local dev environment in slack.

### CODE REVIEW

**Review by AlexsJones — DISMISSED** [review]

**Comment by AlexsJones** [comment]
Thanks @JuHyung-Son ! Can you please resolve conflicts?

**Comment by JuHyung-Son** [comment]
i just did it thanks

**Comment by JuHyung-Son** [comment]
@AlexsJones would you review this?

**Review by AlexsJones — CHANGES_REQUESTED** [review]
Sorry for the delay @JuHyung-Son please make pullsecrets optional - we could have it as a single example in the README also e.g.

```
<details>

<summary>Pull secrets</summary>

...
</details>
```

File: README.md [hunk 1]
```diff
@@ -50,6 +50,8 @@ spec:
   noCache: false
   repository: ghcr.io/k8sgpt-ai/k8sgpt
   version: v0.3.8
+  imagePullSecrets:
```
Comment (AlexsJones) [inline README.md:53]: Not everyone will use pull secrets, even if they do can you put a more generic secret name e.g. `mysecret` ? we only need an example of this in the full example of the `k8sgpt` CR

Comment (JuHyung-Son) [reply README.md]: you are right. i add k8sgpt configuration option

**Review by AlexsJones — APPROVED** [review]

**Comment by JuHyung-Son** [comment]
@AlexsJones i resolved DCO

**Pull request merged**